crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -446,29 +445,32 @@ impl CumulocityConverter {
 
     pub async fn process_health_status_message(
         &mut self,
+        entity: &EntityTopicId,
         message: &Message,
     ) -> Result<Vec<Message>, ConversionError> {
         let mut mqtt_messages: Vec<Message> = Vec::new();
 
         // Send the init messages
-        if check_tedge_agent_status(message)? {
-            create_tedge_agent_supported_ops(self.ops_dir.clone()).await?;
+        if self.is_message_tedge_agent_up(message)? {
+            create_tedge_agent_supported_ops(&self.ops_dir).await?;
             mqtt_messages.push(create_get_software_list_message()?);
         }
 
         // When there is some messages to be sent on behalf of a child device,
         // this child device must be declared first, if not done yet
-        let topic_split: Vec<&str> = message.topic.name.split('/').collect();
-        if topic_split.len() == 4 {
-            let child_id = topic_split[2];
-            add_external_device_registration_message(
-                child_id.to_string(),
-                &mut self.children,
-                &mut mqtt_messages,
-            );
+        if let Some(parent) = entity.default_parent_identifier() {
+            if parent.is_default_child_device() {
+                let child_id = parent.default_device_name().unwrap();
+                add_external_device_registration_message(
+                    child_id.to_string(),
+                    &mut self.children,
+                    &mut mqtt_messages,
+                );
+            }
         }
 
         let mut message = convert_health_status_message(
+            entity,
```

Comment:
Before, calling `convert_health_status_message()`, the converter must use its `self.entity_store` to get the `EntityMetadata` attached to the entity topic id. This metadata and not the topic identifier have to be passed to `convert_health_status_message()`.